index.js

```diff
@@ -0,0 +1,42 @@
+module.exports = function(db, credentials, initialRefreshToken) {
+const knex = require('knex')(db),
+      SimpleOAuth2 = require('simple-oauth2'),
+      oauth2 = SimpleOAuth2.create(credentials);
+
+  var migrated;
+  function migrate() {
+    if (migrated) return migrated;
+    migrated = knex.migrate.latest({ directory: __dirname + '/migrations' });
+    return migrated;
+  }
+
+  function latestToken(type) {
+    return migrate().then(function() {
+      return knex(type + '_tokens').orderBy('id', 'desc').first(type + '_token').then(function(result) {
+        if (result) return result[type + '_token'];
+      });
+    });
+  }
+
+  function refresh() {
```

Comment:
I was originally thinking you would have some kind of id you pass in so that we can easily use this for different auth users.  As it is... yes you can create multiple instances of this thing with the construction function but they are all going to be backed by the same db so you will only ever get the most recent of whichever refresh token was last refreshed.  If you start refreshing with some other user things are going to get confusing and stop working.

Maybe something more like...

```javascript
module.exports = function(db, credentials) {
function refresh(id, refreshToken) {
  return migrate().then(function() {
    return refreshToken || latestToken(id, 'refresh');
  }).then(function(refreshToken) {
    //...
  });
}
```